AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/InitialSync/InitialSyncOperation.swift

```diff
@@ -54,7 +54,19 @@ final class InitialSyncOperation: AsynchronousOperation {
         }
 
         let lastSyncMetadata = getLastSyncMetadata()
-        lastSyncTime = lastSyncMetadata?.lastSync
+        if let lastSync = lastSyncMetadata?.lastSync {
+            let lastSyncDate = Date(timeIntervalSince1970: TimeInterval(lastSync) / 1_000)
+            let secondsSinceLastSync = (lastSyncDate.timeIntervalSinceNow * -1)
+            if secondsSinceLastSync < 0 {
```

Comment:
Maybe just `if lastSyncDate > Date() { ... assume base query ... }` ?

Reply:
Looked at doing this, but I need `secondsSinceLastSync` a couple lines down to determine if we should do a delta query.